libsolidity/formal/PredicateInstance.cpp

```diff
@@ -36,14 +36,23 @@ smtutil::Expression interfacePre(Predicate const& _pred, ContractDefinition cons
 
 smtutil::Expression interface(Predicate const& _pred, ContractDefinition const& _contract, EncodingContext& _context)
 {
-	auto& state = _context.state();
+	auto const& state = _context.state();
 	vector<smtutil::Expression> stateExprs{state.thisAddress(0), state.abi(0), state.crypto(0), state.state()};
 	return _pred(stateExprs + currentStateVariables(_contract, _context));
 }
 
-smtutil::Expression nondetInterface(Predicate const& _pred, ContractDefinition const& _contract, EncodingContext& _context, unsigned _preIdx, unsigned _postIdx)
-{
+smtutil::Expression nondetInterface(
+	Predicate const& _pred,
+	ContractDefinition const& _contract,
+	EncodingContext& _context,
+	smtutil::Expression const& _error,
```

Comment:
Why is `error` a function parameter? You can get it from `state.error()`

Reply:
I wanted an option to pass something else.
Basically, I believe we need to start the `nondet_interface`s with `error == 0`.
So now, I am adding a unit clauses like `(nondet_interface_C_54_0  0  this_0 abi_0 crypto_0 state_0  state_0)`.
For that I need this additional parameter.
I guess this could be changed to something like
```
(=>
  error_0 == 0
  (nondet_interface_C_54_0  error_0  this_0 abi_0 crypto_0 state_0  state_0)
)
```
and than I can get rid of the extra parameter in `PredicateInstance::nondetInterface`.
Would that be preferable?

Reply:
Right. I personally would prefer that, but don't have a strong opinion. Also fine if you prefer the current way.

Reply:
I'll check how big a change this would be.

Reply:
I have updated the code so that the extra parameter is not needed anymore and the encoding now looks like the latter example above.